functions.php

```diff
@@ -59,11 +59,16 @@ function as_schedule_cron_action( $timestamp, $schedule, $hook, $args = array(),
 }
 
 /**
- * Cancel the next occurrence of a job.
+ * Cancel the next occurrence of a scheduled action.
+ *
+ * In the case of a recurring or cron action, this will also avoid all future occurrences
```

Comment:
No it belongs here.

It's here to clarify a gotcha - while only the next recurring or cron action is unscheduled by this method, that will also prevent all future instances of that recurring or cron action from being run.

This is because recurring and cron actions are scheduled in a sequence instead of all being scheduled at once (that is for various reasons, not the least of which is that it's at best inefficient, and at worst impossible, to schedule all actions for a recurring action sequence, as it has no end date, e.g. for a daily recurring action, you'd need to schedule >=365 / year for an indefinite number of years, potentially 10,000s that may never be run). So for those actions, unscheduling just the one, next recurring or cron action will also terminate the entire sequence, which is functionally equivalent to all unscheduled. That's is the intended (and existing) behaviour, but its also potentially unexpected behaviour, which is why I'm documenting it here.

I've updated the docblock to try to clarify that. New docblock:

```
 * While only the next instance of a recurring or cron action is unscheduled by this method, that will also prevent
 * all future instances of that recurring or cron action from being run. Recurring and cron actions are scheduled in
 * a sequence instead of all being scheduled at once. Each successive occurrence of a recurring action is scheduled
 * only after the former action is run. If the next instance is never run, because it's unscheduled by this function,
 * then the following instance will never be scheduled (or exist), which is effectively the same as being unscheduled
 * by this method also.
```

Let me know if that's still unclear!

I don't see a need to be explicit about all recurring actions being unscheduled by the _all_ method, because that should be self explanatory by the function name.